[aureliony] iP

aureliony opened the pull request:
[![CI Status](https://github.com/aureliony/ip/workflows/Java%20CI/badge.svg)](https://github.com/aureliony/ip/actions)

# ConvoBot
> "Don't be afraid to give up the good to go for the great." — John D. Rockefeller ([source](https://www.goodreads.com/quotes/257630-don-t-be-afraid-to-give-up-the-good-to-go))

ConvoBot elevates your task management to the next level.

* text-based
* easy to learn
* ~FAST~ _SUPER FAST_ to use

All you need to do is,

1. download it from [here](https://github.com/aureliony/ip/releases).
2. run it.
3. add your tasks.
4. let it manage your task for you 😉

And it is **FREE**!

Features:

* [x]  Manage todos
* [x]  Manage deadlines
* [x]  Manage events
* [ ]  Manage reminders
* [x]  Graphical UI
* [ ]  Cloud storage

It is written in Java. Here's the `main` method:

```java
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

aureliony committed:
Implement Level-0

aureliony committed:
Refactor welcome and exit strings

aureliony committed:
Refactor welcome and exit strings

aureliony committed:
Refactor welcome and exit strings into functions

aureliony committed:
Implement Level-1

aureliony committed:
Add filetypes to gitignore

aureliony committed:
Remove .class files

aureliony committed:
Implement Level-2

aureliony committed:
Implement Level-3 and A-Classes

aureliony committed:
Implement Level-4 and A-Inheritance

aureliony committed:
Implement A-TextUiTesting

aureliony committed:
Implement Level-5 and A-Exceptions

aureliony committed:
Implement Level-6 and A-Collections

aureliony committed:
Refactor delete operation

aureliony committed:
Update runtest.bat to match new chatbot name

aureliony committed:
Change constant variable name to match conventions

aureliony committed:
Implement Level-7

aureliony committed:
Merge branch 'branch-Level-7'

aureliony committed:
Implement Level-8

aureliony committed:
Merge branch 'branch-Level-8'

aureliony committed:
Implement A-MoreOOP and A-Packages

aureliony committed:
Merge remote-tracking branch 'origin/add-gradle-support'

aureliony committed:
Implement A-Gradle

aureliony committed:
Make several variables private

aureliony committed:
Implement A-JUnit and A-Libraries

aureliony committed:
Implement A-Jar

aureliony committed:
Implement A-JavaDoc and convert CRLF line endings to LF

aureliony committed:
Implement A-CodingStandard

aureliony committed:
Implement Level-9

aureliony committed:
Merge branch 'branch-A-JavaDoc'

aureliony committed:
Resolve merge conflicts

aureliony committed:
Resolve merge conflicts and implement A-Streams

aureliony committed:
Fix a bug with the Delete command

aureliony committed:
Implement A-AbstractClasses and add stubbing to JUnit tests

aureliony committed:
Remove unnecessary import

aureliony committed:
Update .gitignore to ignore all files generated by ConvoBot

aureliony committed:
Fix bug with Storage.write()

aureliony committed:
Remove file creation from Storage.read()

aureliony committed:
Move all classes to convobot package

aureliony committed:
Add javafx dependency to build.gradle

aureliony committed:
Add JavaFX dependency and rename UI to Ui

aureliony committed:
Add FXML dependency and bump JavaFX version

aureliony committed:
Move ConvoBot.java to cli folder and add GUI icons

aureliony committed:
Split ConvoBot.java into CLI and GUI

aureliony committed:
Implement Level-10 and drop CLI support

aureliony committed:
Remove ConvoBotTest.java

aureliony committed:
Fix checkstyle errors

aureliony committed:
Merge branch 'branch-Level-10'

aureliony committed:
Update Javadoc for ResponseConstructor.java

aureliony committed:
Remove text-ui-test

aureliony committed:
Fix dialog text being cut off if too long

aureliony committed:
Fix padding issues

aureliony committed:
Implement styling with dark theme and custom fonts

aureliony committed:
Force window to be un-resizable

getsquared commented on src/main/java/convobot/utils/DateTime.java:
```diff
@@ -0,0 +1,55 @@
+ */
+public class DateTime {
+
+    private static final DateTimeFormatter inputFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
```
Constant name (static final) could be uppercase and use underscores instead? Like INPUT_FORMATTER.

aureliony replied:
Accepted.

getsquared commented on src/main/java/convobot/tasks/Task.java:
```diff
@@ -0,0 +1,91 @@
+public abstract class Task {
+
+    private String description;
+    private boolean isDone;
```
I like that all your booleans are named like booleans 👍

getsquared commented on src/main/java/convobot/utils/Storage.java:
```diff
@@ -0,0 +1,70 @@
+ */
+public class Storage {
+
+    private final String filePath;
```
Could this be named in uppercase as a constant? Like FILE_PATH?

aureliony replied:
Accepted.

aureliony replied:
Rejected. filePath is not a static variable.

getsquared commented on src/main/java/convobot/commands/Mark.java:
```diff
@@ -0,0 +1,43 @@
+ */
+public class Mark implements Command {
+
+    private final int i;
```
This is not a temporary or iterator variable. It could have a longer name rather than just i?

aureliony replied:
Accepted, renamed to indexToMark.

getsquared commented on src/main/java/convobot/utils/Parser.java:
```diff
@@ -0,0 +1,190 @@
+        int k = -1;
+        String description;
+
+        switch (commandType) {
```
I like that your switch statements have proper indentation. Keep it up 👍

getsquared reviewed (commented):
Looks good to me 👍 Just minor naming changes.

gosongying commented on src/main/java/convobot/commands/Add.java:
```diff
@@ -0,0 +1,42 @@
+package commands;
+
+import tasks.Task;
```
Perhaps can add a blank below to separate different packages? Same issues in several places too.

aureliony replied:
Rejected. Adding a blank line causes checkstyle to fail.

gosongying commented on src/main/java/convobot/commands/Delete.java:
```diff
@@ -0,0 +1,44 @@
+ */
+public class Delete implements Command {
+
+    private final int i;
```
Maybe can make the variable name more meaningful like index because the scope is not small. Same issue in different places too.

aureliony replied:
Accepted, renamed to indexToDelete.

gosongying commented on src/main/java/convobot/utils/DateTime.java:
```diff
@@ -0,0 +1,55 @@
+ */
+public class DateTime {
+
+    private static final DateTimeFormatter inputFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
```
Shouldn't constant name be all uppercase using underscore to separate words?

aureliony replied:
Accepted.

gosongying commented on src/main/java/convobot/utils/Parser.java:
```diff
@@ -0,0 +1,190 @@
+            commandType = CommandType.valueOf(inputList.get(0).toUpperCase());
+        } catch (IllegalArgumentException e) {
+            String strs = Arrays.stream(CommandType.values())
+                                .map(Enum::name)
```
Nice indentation here!

gosongying commented on src/main/java/convobot/utils/ResponseConstructor.java:
```diff
@@ -0,0 +1,132 @@
+    /**
+     * Displays a message indicating that a task has been removed.
+     *
+     * @param taskString the string representation of the removed task
+     * @param dbSize     the size of the task list after removing the task
+     */
+    public void showRemoved(String taskString, int dbSize) {
+        response.append("Noted. I've removed this task:\n");
+        response.append("   " + taskString + "\n");
+        response.append("Now you have " + Integer.toString(dbSize) + " tasks in the list.\n");
+    }
+
+    /**
+     * Displays a message indicating that a task has been marked as done.
+     *
+     * @param taskString the string representation of the marked task
+     */
+    public void showMarked(String taskString) {
+        response.append("Nice! I've marked this task as done:\n");
+        response.append("   " + taskString + "\n");
+    }
+
+    /**
+     * Displays a message indicating that a task has been marked as not done.
+     *
+     * @param taskString the string representation of the unmarked task
+     */
+    public void showUnmarked(String taskString) {
+        response.append("OK, I've marked this task as not done yet:\n");
+        response.append("   " + taskString + "\n");
+    }
+
+    /**
+     * Displays an error message in the console.
+     *
+     * @param errorMsg the error message to be displayed
+     */
+    public void showError(String errorMsg) {
+        response.append(errorMsg + "\n");
+    }
```
Good use of white space within statements.

gosongying reviewed (commented):
LGTM. Just a few nits to fix such as constant name and separation among import statements for different packages.

aureliony committed:
Implement peer review suggestions

aureliony committed:
Implement A-Streams

Added comment to explain use of Java streams in Parser.java

aureliony committed:
Implement A-Assertions

Improved the code by adding assertions in various locations.

Enabled assertions in build.gradle run task.

Asserted that ConvoBot's response string should not be empty.

Asserted false in a code block that is not meant to be reachable under any circumstances.

aureliony committed:
Standardized the code

Added a blank line after each class declaration

aureliony committed:
Implement A-CodeQuality

Improved the code quality according to topic W5.4.

Refactored TaskList.java by extracting the enforceArrayBounds() method.

aureliony committed:
Added assertation in Parser.java

aureliony committed:
Refactor parseTaskFromLine() in Parser.java

Refactored parseTaskFromLine() by defining constants at the start of the method

aureliony committed:
Refactor Parser.java

Separated Parser.java into TaskParser.java and CommandParser.java.

Separated the corresponding test class ParserTest.java into TaskParserTest.java and CommandParserTest.java as well.

aureliony committed:
Refactor Event.java

aureliony committed:
Merge pull request #1 from aureliony/branch-A-Assertions

Merge branch-A-Assertions into master

aureliony committed:
Merge pull request #2 from aureliony/branch-A-Streams

Implement A-Streams

aureliony committed:
Merge branch 'master' into branch-A-CodeQuality

aureliony committed:
Refactor CommandParser.java

aureliony committed:
Merge pull request #7 from aureliony/branch-A-CodeQuality

Improve code quality

aureliony committed:
Fix compilation error with CommandParser.java

aureliony committed:
Implement A-CI

aureliony committed:
Bump CI dependencies

aureliony committed:
Implement C-BetterSearch

aureliony committed:
Merge branch 'branch-A-BetterSearch' into master

aureliony committed:
Refactor build.gradle

aureliony committed:
Move all packages into cvb package

aureliony committed:
Trim whitespaces and improve storage reading

aureliony committed:
Refactor Storage.java

aureliony committed:
Refactor TaskParser.java

aureliony committed:
Modify Javadocs to follow rules

aureliony committed:
Update README.md

aureliony committed:
Refactor CommandParser.java

aureliony committed:
Add Javadocs for CommandType.java

aureliony committed:
Fix scroll wheel not working

aureliony committed:
Add workflow badge

aureliony committed:
Update README.md

aureliony committed:
Remove traces of Duke

aureliony committed:
Merge remote-tracking branch 'refs/remotes/origin/master'

aureliony committed:
Cleanup build.gradle

aureliony committed:
Fix | character causing issues

aureliony committed:
Validate event /from appears before /to

aureliony committed:
Add welcome message and refactor MainWindow.java

aureliony committed:
Remove unused import

aureliony committed:
Add Ui.png

aureliony committed:
Add user guide

aureliony committed:
Update README.md

aureliony committed:
Update README.md

aureliony committed:
Change date display format

aureliony committed:
Cleanup Javadocs and rename some methods

aureliony committed:
Update README.md

aureliony committed:
Make window resizable

aureliony committed:
Make profile pictures rounded

aureliony committed:
Update Ui.png

aureliony committed:
Add Java version compatibility to build.gradle

aureliony committed:
Simplify JavaFX dependency in build.gradle

aureliony committed:
Remove magic number in Main.java

aureliony committed:
Update README.md

aureliony committed:
Add GPL 2.0 LICENSE

aureliony committed:
Fix .jar not working on linux/mac

aureliony committed:
Reorder JavaFX dependencies

aureliony committed:
Move tests to cvb folder
